bpfd/src/bpf.rs

```diff
@@ -333,9 +337,88 @@ impl BpfManager {
                     Err(BpfdError::UnableToPinProgram(e))
                 })?;
 
+            Ok(id)
+        } else if let Program::Uprobe(ref program) = p {
+            let map_pin_path = format!("{RTDIR_FS_MAPS}/{id}");
+            fs::create_dir_all(map_pin_path.clone())
+                .await
+                .map_err(|e| BpfdError::Error(format!("can't create map dir: {e}")))?;
+
+            let program_bytes = if program
+                .data
+                .path
+                .clone()
+                .contains(BYTECODE_IMAGE_CONTENT_STORE)
+            {
+                get_bytecode_from_image_store(program.data.path.clone()).await?
+            } else {
+                read(program.data.path.clone()).await?
+            };
+
+            let mut loader = BpfLoader::new();
+
+            for (name, value) in &program.data.global_data {
+                loader.set_global(name, value.as_slice());
+            }
+
+            let mut loader = loader
+                .map_pin_path(map_pin_path.clone())
+                .load(&program_bytes)?;
+
+            let uprobe: &mut UProbe = match loader.program_mut(&program.data.section_name) {
+                Some(p) => p.try_into(),
+                None => {
+                    let _ = fs::remove_dir_all(map_pin_path).await;
+                    return Err(BpfdError::SectionNameNotValid(
+                        program.data.section_name.clone(),
+                    ));
+                }
+            }?;
+
+            uprobe.load()?;
+            p.save(id)
+                .map_err(|_| BpfdError::Error("unable to persist program data".to_string()))?;
+
+            let link_id = uprobe
+                .attach(
+                    program.info.fn_name.as_deref(),
+                    program.info.offset.unwrap_or_default(),
+                    program.info.target.clone(),
+                    program.info.pid,
+                )
+                .or_else(|e| {
+                    p.delete(id).map_err(|_| {
+                        BpfdError::Error(
+                            "new dispatcher cleanup failed, unable to delete program data"
```

Comment:
This comment is wrong i think :) 

There's no dispatcher here!

Reply:
Yup.  That was a copy-pasted from the tracepoint code (which I assume was copy-pasted from the multi-attach code).  I'll fix both places.